addons/project/models/res_partner.py

```diff
@@ -80,3 +80,30 @@ def action_view_tasks(self):
         else:
             action['domain'] = search_domain
         return action
+
+    @api.model
+    def get_mention_suggestions_from_task(self, task_id, search, limit=8):
+        """Return 'limit'-first partners' such that the name or email matches a 'search' string.
+        Prioritize partners that are also (internal) users, and then extend the research to all partners.
+        Only followers of the given task or followers of its project are returned.
+        The return format is a list of partner data (as per returned by `mail_partner_format()`).
+        """
+        task_sudo = self.env['project.task'].sudo().search([('id', '=', task_id)])
+        shared_project_sudo = task_sudo.project_id
+        shared_project = shared_project_sudo.with_user(self.env.user)
+        if not (shared_project and shared_project._check_project_sharing_access()):
+            return []
+        try:
+            shared_project.check_access_rights('read')
+            shared_project.check_access_rule('read')
+        except AccessError:
+            return []
```

Comment:
Actually, we can avoid raising an exception, by using the second parameter defined in `check_acces_rights` and use `_filter_access_rule`. 🙂 

```suggestion
        if not (
            project 
            and project._check_project_sharing_access()
            and project.check_access_rights('read', raise_exception=False)
            and project._filter_access_rules_python('read')
        ):
            return []
```

ps: I did not test my suggestion but it should be equivalent to yours

Reply:
I would say yes because if the user cannot read the project then he/she cannot enter in Project sharing feature. 🙂